docs/backup/backup-rsync.md

```diff
@@ -0,0 +1,180 @@
+
+```bash
+ssh_key='/your/private/key/location/id_rsa'
+remote_port='xxxx'
+```
+Replace the strings with your port/private key location.
+
+Now, replace the user/server domain and issue the rsync command for a dry run
+(nothing will be copied yet):
+
+```bash
+rsync -avzP --dry-run --delete --human-readable --stats -e  "ssh -p $remote_port -i $ssh_key" root@example.com:/srv/seafile/haiwen/seafile-data/ ./
+```
+
+NOTE: Preserving the `.../seafile-data/` trailing slash means that we wish to
+copy only the directory contents, not the directory itself.
+
+The arguments are as follows:
+
+```
+ -a, --archive      archive mode (preserves permissions)
+ -v, --verbose      increase verbosity
+ -z, --compress     compress file data during the transfer
+ -P                 show progress
+ -n, --dry-run      doesn't make any changes
+ --delete           delete extraneous local files
+ --human-readable    output in a human-readable format
+ --stats             give some file-transfer stats
+ -e                  alternative remote shell
+```
+
+The output should look something like this:
+```
+Number of files: 89,298 (reg: 80,773, dir: 8,525)
+Number of created files: 4,803 (reg: 4,345, dir: 458)
+Number of deleted files: 0
+Number of regular files transferred: 4,349
+Total file size: 86.94G bytes
+Total transferred file size: 23.91G bytes
+Literal data: 0 bytes
+Matched data: 0 bytes
+File list size: 1.66M
+File list generation time: 0.001 seconds
+File list transfer time: 0.000 seconds
+Total bytes sent: 26.16K
+Total bytes received: 4.57M
+
+sent 26.16K bytes  received 4.57M bytes  612.17K bytes/sec
+total size is 86.94G  speedup is 18,934.93 (DRY RUN)
+```
+The `Total file size: 86.94G bytes` and
+`Total transferred file size: 23.91G bytes` are interesting to look at. The
+first tells us that the remote directory is 86.96 GB in size, but the second
+tell us that only 23.91 GB will be copied to our local drive. Herein lies the
+advantage of rsync: because in our example the files were already synchronized
+beforehand, this new sync will only pull the modified data, not all of it. If
+you run the sync for the first time, however, you will have to download all
+the files with their full size.
+
+When you're ready to begin synching, issues the above command without the
+--dry-run parameter.
+
+```bash
+rsync -avzP --delete --human-readable --stats -e  "ssh -p $remote_port -i $ssh_key" root@example.com:/srv/seafile/haiwen/seafile-data/ ./
+```
+
+### Conclusion
+
+If everything went according to plan, we have retrieved a copy of the
+`seafile`, `seahub` and `ccnet` databases and placed them in
+`/run/media/foo/seafbackup/seafile_databases` .
+
+The contents of the `seafile_data` folder are now synchronized to
+`/run/media/foo/seafbackup/seafile_databases/` .
+
+Recovering from a data loss will just be a matter of restoring the databases
+and the data directory.
+
+As the sync will only pull the modified files from the remote host, the more
```

Comment:
Backing up to a remote host with an unprivileged user and automatic snapshots significantly increases security of the backups. An attacker cannot delete the backup without intruding into the backup host as well, crypto trojans cannot encrypt the backup, some history can be kept to be be prepared for issues that are found late and depending on the target location there is great protection against lightning strikes, other bad weather conditions, fire and theft.